EIPS/eip-network_upgrade_windows.md

```diff
@@ -0,0 +1,171 @@
+> recommendation and are easily modifiable prior to final call. They thread the
+> needle between many third quarter and fourth quarter holidays.
+
+A roadmap upgrade typically should not occur in adjacent upgrade windows. Hence
+if a roadmap upgrade occurred in April then the July window would not be used
+for network upgrades.
+
+If a planned roadmap upgrade needs to be rescheduled then strong consideration
+should be given to rescheduling the upgrade for the next window in three months
+time.
+
+Not every upgrade window must be used.
+
+### Priority Network Upgrades
+
+Priority network upgrades are reserved for upgrades that require more urgency
+than a roadmap network upgrade yet do not present a systemic risk to the network
+or the ecosystem. To date there have been no examples of a priority upgrade.
+Possible examples may include roadmap upgrades that need to occur in multiple
+upgrades or for security risks that have a existing mitigation in place that
+would be better served by a network upgrade. Another possible reason may be to
+defuse the difficulty bomb due to postponed roadmap upgrades.
+
+Priority network upgrades are best launched in unused roadmap launch windows,
+namely the third week of January, April, July, and October. If necessary they
+may be launched in the third week of any month, but strong consideration and
+preference should be given to unused roadmap launch windows.
+
+Priority network upgrades should be announced and a block chosen far enough in
+advance so major clients implementors can release software with the needed block
+number in a timely fashion. It is expected that two to four weeks would be
+sufficient.
+
+### Critical Network Upgrades
+
+Critical network upgrades are network upgrades that are designed to address
+systemic risks to the protocol or to the ecosystem. Historical examples include
+Dao Fork, Tangerine Whistle, and Spurious Dragon.
+
+This EIP provides neither guidance nor restrictions to the development and
+deployment of these emergency hard forks. These upgrades are typically launched
+promptly after a solution to the systemic risk is agreed upon between the client
+implementors.
+
+### Network Upgrade Naming
+
+The choice of a name for a network upgrade can help communicate what type of
+network upgrade it is. There have been patterns used in the past that if
+continued would help communicate this.
+
+Roadmap Network Upgrades have historically been given single word names. The
+first three network upgrades were Frontier, Homestead, and Metropolis.
+Metropolis was split into two upgrades, Byzantium and Constantinople. As of the
+writing of this EIP the next network upgrade will be called Istanbul.
+
+Critical Network Upgrades have historically been two word phrases. Sometimes the
+name applies to the issue at hand, sometimes it do not. Previous Critical
+Network upgrade names have been Dao Fork, Tangerine Whistle, Spurious Monkey,
+and Constantinople Fix.
```

Comment:
Maybe naming is out of scope? After all, end users only need to know the target version.

I would also push for a more explicit versioning scheme alongside the naming convention, but that is also probably outside the scope here.

Reply:
Agreed, out of scope.  It belongs in a Roadmap Cadence EIP.  Deleting section.